sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2150,6 +2150,15 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
     // std::array<std::byte, 16>. For details about this extension,
     // see sycl/doc/extensions/IntelGPU/IntelGPUDeviceInfo.md.
     return ReturnValue(Device->ZeDeviceProperties->uuid.id);
+  case PI_DEVICE_INFO_ATOMIC_64: {
+    if (Device->ZeDeviceModuleProperties->flags &
+        ZE_DEVICE_MODULE_FLAG_INT64_ATOMICS) {
+      bool result = true;
+      std::memcpy(ParamValue, &result, sizeof(bool));
+      return PI_SUCCESS;
+    }
+    return PI_INVALID_VALUE;
+  }
```

Comment:
Shouldn't we return false as a parameter value and return PI_SUCESS if Atomic64 is not supported?

Reply:
In this case we check if device can perform 64-bit atomic operations. As I know, L0 device can perform them if ZE_DEVICE_MODULE_FLAG_INT64_ATOMICS is set, whereas on OpenCL we need to check for two extensions. 
As a result, we return PI_SUCCESS (piDeviceGetInfo passed without errors) and make a copy of true to ParamValue.

Reply:
Sorry, but my question is not about the case when ZE_DEVICE_MODULE_FLAG_INT64_ATOMICS is set but about the case when this flag is not set. Why we need to return PI_INVALID_VALUE if ZE_DEVICE_MODULE_FLAG_INT64_ATOMICS is not set? This is valid case and we just need to return false as a result, using your code:
`      bool result = false;
      std::memcpy(ParamValue, &result, sizeof(bool));  return PI_SUCCESS;` 
Now PI_INVALID_VALUE is returned and this will result in exception being thrown in the SYCL RT just because 64-bit atomic operations are not supported and this is wrong.

Reply:
Okay, I understand. Commit with fix is on the way!